addon/services/prefetch.js

```diff
@@ -25,14 +25,16 @@ if (gte('3.6.0')) {
             let changeSet = createPrefetchChangeSet(privateRouter, transition);
             if (changeSet.shouldCall) {
               for (let i = 0; i < changeSet.for.length; i++) {
-                let { route, fullParams } = changeSet.for[i];
-                if (seenRoutes.has(route)) continue;
-
-                  route._prefetched = new RSVP.Promise(r => {
-                    return r(route.prefetch(fullParams, transition))
+                // normalize route to always be a promise since lazy loaded
+                // routes are promises, while preloaded routes are not. We want
+                // to treat them both the same.
+                RSVP.resolve(changeSet.for[i]).then(({ route, fullParams }) => {
+                  RSVP.resolve(route).then(resolvedRoute => {
```

Comment:
We should use `RSVP.Promise.resolve()` not `RSVP.resolve()` here.

```suggestion
                  RSVP.Promise.resolve(route).then(resolvedRoute => {
```